aws/resource_aws_serverlessapplicationrepository_cloudformation_stack.go

```diff
@@ -0,0 +1,379 @@
+	}
+
+	log.Printf("[DEBUG] Creating Serverless Application Repository change set: %s", changeSetRequest)
+	changeSetResponse, err := serverlessConn.CreateCloudFormationChangeSet(&changeSetRequest)
+	if err != nil {
+		return fmt.Errorf("error creating Serverless Application Repository change set (%s): %w", stackName, err)
+	}
+
+	d.SetId(aws.StringValue(changeSetResponse.StackId))
+
+	_, err = cfwaiter.ChangeSetCreated(cfConn, d.Id(), aws.StringValue(changeSetResponse.ChangeSetId))
+	if err != nil {
+		return fmt.Errorf("error waiting for Serverless Application Repository change set (%s) creation: %w", stackName, err)
+	}
+
+	log.Printf("[INFO] Serverless Application Repository change set (%s) created", d.Id())
+
+	requestToken := resource.UniqueId()
+	executeRequest := cloudformation.ExecuteChangeSetInput{
+		ChangeSetName:      changeSetResponse.ChangeSetId,
+		ClientRequestToken: aws.String(requestToken),
+	}
+	log.Printf("[DEBUG] Executing Serverless Application Repository change set: %s", executeRequest)
+	_, err = cfConn.ExecuteChangeSet(&executeRequest)
+	if err != nil {
+		return fmt.Errorf("Executing Serverless Application Repository change set failed: %w", err)
+	}
+
+	_, err = cfwaiter.StackCreated(cfConn, d.Id(), requestToken, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return fmt.Errorf("error waiting for Serverless Application Repository CloudFormation Stack (%s) creation: %w", d.Id(), err)
+	}
+
+	log.Printf("[INFO] Serverless Application Repository CloudFormation Stack (%s) created", d.Id())
+
+	return resourceAwsServerlessApplicationRepositoryCloudFormationStackRead(d, meta)
+}
+
+func resourceAwsServerlessApplicationRepositoryCloudFormationStackRead(d *schema.ResourceData, meta interface{}) error {
+	serverlessConn := meta.(*AWSClient).serverlessapplicationrepositoryconn
+	cfConn := meta.(*AWSClient).cfconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	applicationID := d.Get("application_id").(string)
+	semanticVersion := d.Get("semantic_version").(string)
+	descriptor := applicationID
+	if semanticVersion != "" {
+		descriptor += fmt.Sprintf(", version %s", semanticVersion)
+	}
+
+	getApplicationOutput, err := finder.Application(serverlessConn, applicationID, semanticVersion)
+	if err != nil {
+		return fmt.Errorf("error getting Serverless Application Repository application (%s): %w", descriptor, err)
+	}
+
+	if getApplicationOutput.Version == nil {
+		return fmt.Errorf("error getting Serverless Application Repository application (%s): returned empty version record", descriptor)
+	}
+
+	version := getApplicationOutput.Version
+	d.Set("semantic_version", version.SemanticVersion)
+
+	parameterDefinitions := flattenServerlessRepositoryParameterDefinitions(version.ParameterDefinitions)
+
+	stack, err := cffinder.Stack(cfConn, d.Id())
+	var e *resource.NotFoundError
+	if errors.As(err, &e) {
```

Comment:
Since its not being used yet (only `tfresource.TimedOut`), it might be good time to update https://github.com/hashicorp/terraform-provider-aws/blob/2975c11f8207d35728c349c0f7ddcde8927b5e52/aws/internal/tfresource/errors.go#L9-L12 with `errors.As` and use it here. 👍